### DISCUSSION TITLE
Is there a way a can map fontWeight to custom fonts (like Dripsy package) ?

### DISCUSSION THREAD

**Discussion by ammarfaris** [jpudysz/react-native-unistyles#111] [Q&A] [answered]
Hi again, 

I am trying to add an external font (`Geist` font - https://github.com/vercel/geist-font )  in my solito app following this solito starter template ( https://github.com/nandorojo/solito/tree/master/example-monorepos/with-custom-font ).

In the download package of `Geist` font, vercel included a [variable-version](https://fonts.google.com/knowledge/introducing_type/introducing_variable_fonts)  `GeistVariableVF.ttf `; somehow the variation of `fontWeight` from '100 'to' 900' only works with `nextjs web` ; the range does not work well with `expo mobile` unfortunately.

my question is then , if i revert back to the old way of adding fonts with separate "weights" like adding  `Geist-Bold.otf` + `Geist-Black.otf` + and more , is there a way to map the fontWeight to these custom individual font variations???

In the solito start with `dripsy` and `custom-font` example, [this is the way](https://github.com/nandorojo/solito/blob/master/example-monorepos/with-custom-font/packages/app/provider/dripsy.tsx) `drispy` allow us to map the fonts :

```js
import { DripsyProvider, makeTheme } from 'dripsy'
import { Platform } from 'react-native'

const rootFontName = 'Inter-Regular'

const webFont = (font: string) =>
  Platform.select({
    web: `${font}, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Helvetica, Arial, Inter-serif`,
    default: font,
  })

const theme = makeTheme({
  // https://www.dripsy.xyz/usage/theming/create
  text: {
    p: {
      fontSize: 16,
    },
  },
  customFonts: {
    [rootFontName]: {
      '400': webFont(rootFontName),
      '500': webFont(rootFontName),
      600: webFont('Inter-Bold'),
      700: webFont('Inter-Bold'),
      800: webFont('Inter-Black'),
      900: webFont('Inter-Black'),
      bold: webFont('Inter-Bold'),
      default: webFont(rootFontName),
      normal: webFont(rootFontName),
    },
  },
  fonts: {
    root: rootFontName,
  },
})

export function Dripsy({ children }: { children: React.ReactNode }) {
  return (
    <DripsyProvider
      theme={theme}
      // this disables SSR, since react-native-web doesn't have support for it (yet)
      ssr
    >
      {children}
    </DripsyProvider>
  )
}

```

Thanks @jpudysz and the contributors for an awesome library!

**Comment by jpudysz** [comment]
Hey, of course you can. I would create a plugin: https://reactnativeunistyles.vercel.app/reference/plugins/

Something like this:
```tsx
export const fontWeightPlugin: UnistylesPlugin = {
    name: 'fontWeightPlugin',
    onParsedStyle: (_key, styles) => {
        if ('fontWeight' in styles) {
             switch (styles.fontWeight) {
                case '400':
                case '500': {
                   styles.fontFamily = 'Inter-Regular'
                }
                 // other cases
            }
        }
        
        return styles
    }
}
```

**Reply by jpudysz** [reply]
I added it to the docs: https://reactnativeunistyles.vercel.app/reference/plugins/#example

**Reply by ammarfaris** [reply]
Thank you so much @jpudysz for the guide. 

I am trying to add the plugin now ; typescript is telling me that the return type can be `undefined`,  how do i fix this?

![can return undefined](https://github.com/jpudysz/react-native-unistyles/assets/1182205/2d8d64ee-ff98-4ad7-ae98-f0e759858bc3)

**Reply by ammarfaris** [reply]
Hi @jpudysz, I just tried to add the plugin, but the addition of plugin to `UnistylesRegistry`  causes issues to the unistyles setup for the app:

For mobile:
All styles are broken after adding the plugin

For web, I got this error:
![unistyles web error](https://github.com/jpudysz/react-native-unistyles/assets/1182205/7fc5c6b0-5db0-4aad-afe0-413068fe5aaa)

Do you have a tested working example?

**Reply by jpudysz** [reply]
Hey, I don't know what's wrong whith your code as you didn't show the example, but simply parse the style and return new styles:

```tsx
import type { UnistylesPlugin } from 'react-native-unistyles'

export const fontWeightPlugin: UnistylesPlugin = {
    name: 'fontWeightPlugin',
    onParsedStyle: (_key, styles) => {
        if ('fontWeight' in styles) {
            switch (styles.fontWeight) {
                case '400':
                default: {
                    styles.fontFamily = 'Inter-Regular'
                }
            }
        }

        return styles
    }
}
```

Result:

<img width="360" alt="image" src="https://github.com/jpudysz/react-native-unistyles/assets/9088288/0121057b-ff9a-4a29-bd2e-635e4957027d">

Just don't overcomplicate it. Function takes styles, do whatever you want with them and return styles (object),

**Reply by ammarfaris** [reply, marked as answer]
sorry my bad that the earlier code shown was hidden by the typescript error box, 
I was mistakenly returning (`return`) on each `switch` case, whereby it should be `break`

This works now without error:

```js

import { UnistylesPlugin } from 'react-native-unistyles'

export const fontWeightPlugin: UnistylesPlugin  = {
  name: 'fontWeightPlugin',
  onParsedStyle: (_key, styles) => {
    if ('fontWeight' in styles) {
      switch (styles.fontWeight) {
        case '100': {
          styles.fontFamily = 'Geist-Thin'
          break
        }
        case '200': {
          styles.fontFamily = 'Geist-UltraLight'
          break
        }
        case '300': {
          styles.fontFamily = 'Geist-Light'
          break
        }
        case '400': {
          styles.fontFamily = 'Geist-Regular'
          break
        }
        case '500': {
          styles.fontFamily = 'Geist-Medium'
          break
        }
        case '600': {
          styles.fontFamily = 'Geist-SemiBold'
          break
        }
        case '700': {
          styles.fontFamily = 'Geist-Bold'
          break
        }
        case '800': {
          styles.fontFamily = 'Geist-Black'
          break
        }
        case '900': {
          styles.fontFamily = 'Geist-UltraBlack'
          break
        }
        case 'bold': {
          styles.fontFamily = 'Geist-Bold'
          break
        }
        case 'normal': {
          styles.fontFamily = 'Geist-Regular'
          break
        }
        default: {
          styles.fontFamily = 'Geist-Regular'
        }
      }
    }
    return styles
  }
}

```

Thanks!
